js/autotrack/common.js

```diff
@@ -57,9 +87,121 @@ const getFiberNodeComponentHierarchy = currNode => {
     builtinPropExtractorConfig
   );
 
-  return `${getFiberNodeComponentHierarchy(
+  const parentHierarchyRepresentation = getFiberNodeComponentHierarchyTraversal(
     currNode.return
-  )}${elementName};${propsString}|`;
+  );
+
+  parentHierarchyRepresentation.push({
+    elementName,
+    fiberNode: currNode, // Needed to get all props on HeapIgnore components.
+    propsString, // :TODO: Make this an object so we can use it with 'ignoreSpecificProps'
+  });
+
+  return parentHierarchyRepresentation;
+};
+
+// Given an object array representing the component hierarchy from root to target component,
+// traverse through the element list to create the string representation of the hierarchy, taking
+// into account HeapIgnore specifications.
+// Returns an object containing:
+// * hierarchy - the string hierarchy representation of 'hierarchyArray'.
+// * heapIgnoreProps - the final set of HeapIgnore props that applies to the target component. Used
+//     to determine whether to include target text and/or ignore the entire interaction.
+const getHierarchyStringFromTraversal = hierarchyArray => {
+  let currentHeapIgnoreProps = {
```

Comment:
What do you think about encapsulating this in a (ideally Typescript) class? If we wanted to add a new property to this list in the future, we'd have to add it to 4 separate declarations in this file, which seems a bit error-prone.

Reply:
This is definitely a possible refactor, but since this was simplified a bit with the removal of `HeapCaptureRestrictor`, I'd prefer to wait to pull this into its own class for now